pricelist_price_base_custom/models/product_pricelist_item.py

```diff
@@ -0,0 +1,28 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import fields, models
+
+
+class PricelistItem(models.Model):
+    _inherit = "product.pricelist.item"
+
+    base = fields.Selection(
+        selection_add=[("custom_value", "Custom Value")],
+        ondelete={"custom_value": "set default"},
+    )
+
+    def _compute_base_price(self, product, quantity, uom, date, target_currency):
+        """
+        Override to use custom base price if "custom_value"
+        is selected as rule base and custom_base_price is
+        passed in context
+        """
+        if self.base == "custom_value":
+            return (
```

Comment:
IMHO, this is not correct.

A price from a context value is always a bad practice. Don't you have another mean to do so ?

Reply:
> IMHO, this is not correct.
> 
> A price from a context value is always a bad practice. Don't you have another mean to do so ?

hi @rousseldenis, thank you for the feedback!  yup I share the same concerns. However this was the less intrusive to achieve the objective.
Otherwise there would be much more code because of the function param override.